src/main/java/org/fenixedu/academic/domain/ExecutionCourse.java

```diff
@@ -128,6 +128,21 @@ public int compare(ExecutionCourse o1, ExecutionCourse o2) {
 
             };
 
+    public static final Comparator<ExecutionCourse> EXECUTION_COURSE_COMPARATOR_BY_EXECUTION_PERIOD_REVERSED_AND_NAME =
+            new Comparator<ExecutionCourse>() {
+
+                @Override
+                public int compare(ExecutionCourse o1, ExecutionCourse o2) {
+                    final int cep = o2.getExecutionPeriod().compareTo(o1.getExecutionPeriod());
+                    if (cep != 0) {
+                        return cep;
+                    }
+                    final int c = Collator.getInstance().compare(o1.getNome(), o2.getNome());
+                    return c == 0 ? DomainObjectUtil.COMPARATOR_BY_ID.compare(o1, o2) : c;
+                }
+
+            };
+
```

Comment:
This code can be replaced to use `EXECUTION_COURSE_EXECUTION_PERIOD_COMPARATOR` and `EXECUTION_COURSE_NAME_COMPARATOR`. `Comparator.reversed()` should help.
Maybe you can also change the implementation of `EXECUTION_COURSE_COMPARATOR_BY_EXECUTION_PERIOD_AND_NAME` accordingly.

Reply:
Changed both the implementations accordingly.